react/ContactsListModal/DemoProvider.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import contacts from '../ContactsList/data.json'
+import CozyClient from 'cozy-client'
+
+const mockClient = new CozyClient({
+  uri: 'http://cozy.tools:8080',
+  token: 'faketoken'
+})
+
+mockClient.__proto__.requestQuery = () =>
+  Promise.resolve({
+    fetchStatus: 'loaded',
+    data: contacts.map(contact => ({ ...contact, _type: 'io.cozy.contacts' }))
+  })
+
+class Wrapper extends React.Component {
+  getChildContext() {
+    return {
+      client: mockClient
+    }
+  }
+
+  render() {
+    return <>{this.props.children}</>
+  }
+}
```

Comment:
It seems to achieve the same thing as createClientWithData, is there any difference ?

Reply:
Yes the goal is the same. I will keep it as is for now because I already used too much time on this.